cohort-engine-flink-example/src/main/java/com/ibm/cohort/engine/flink/execution/CohortEngineFlinkDriver.java

```diff
@@ -0,0 +1,189 @@
+import org.apache.flink.streaming.api.datastream.DataStreamSource;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.connectors.kafka.FlinkKafkaConsumer;
+import org.apache.flink.streaming.connectors.kafka.FlinkKafkaProducer;
+import org.hl7.fhir.r4.model.MeasureReport;
+
+public class CohortEngineFlinkDriver implements Serializable {
+
+	private static final long serialVersionUID = 1966474691011266880L;
+
+	public static void main(String[] args) throws Exception {
+		ParameterTool params = ParameterTool.fromArgs(args);
+
+		FHIRServerInfo fhirServerInfo = new FHIRServerInfo(
+				params.getRequired("fhirTenantId"),
+				params.getRequired("fhirUsername"),
+				params.getRequired("fhirPassword"),
+				params.getRequired("fhirEndpoint")
+		);
+
+		KafkaInfo kafkaInputInfo = new KafkaInfo(
+				params.getRequired("kafkaBrokers"),
+				params.getRequired("kafkaInputTopic"),
+				params.getRequired("kafkaPassword")
+		);
+
+		KafkaInfo kafkaOutputInfo = null;
+		if (params.has("kafkaOutputTopic")) {
+			kafkaOutputInfo = new KafkaInfo(
+					params.getRequired("kafkaBrokers"),
+					params.getRequired("kafkaOutputTopic"),
+					params.getRequired("kafkaPassword")
+			);
+		}
+
+		CohortEngineFlinkDriver example = new CohortEngineFlinkDriver(
+				fhirServerInfo,
+				params.has("insecureHttpClient")
+		);
+		example.run(
+				params.get("jobName", "cohort-engine"),
+				params.getRequired("kafkaGroupId"),
+				kafkaInputInfo,
+				kafkaOutputInfo,
+				params.has("printOutputToConsole"),
+				params.has("rebalanceInput"),
+				params.has("readFromStart")
+		);
+	}
+
+	private final FHIRServerInfo fhirServerInfo;
+	private final boolean insecureHttpClient;
+
+	private transient MeasureEvaluator evaluator;
+	private transient FhirContext fhirContext;
+	private transient ObjectMapper objectMapper;
+
+	public CohortEngineFlinkDriver(FHIRServerInfo fhirServerInfo, boolean insecureHttpClient) {
+		this.fhirServerInfo = fhirServerInfo;
+		this.insecureHttpClient = insecureHttpClient;
+	}
+
+	private void run(
+			String jobName,
+			String kafkaGroupId,
+			KafkaInfo kafkaInputInfo,
+			KafkaInfo kafkaOutputInfo,
+			boolean printOutputToConsole,
+			boolean rebalanceInput,
+			boolean readFromStart
+	) throws Exception {
+		FlinkKafkaConsumer<String> consumer = new FlinkKafkaConsumer<>(
+				kafkaInputInfo.getTopic(),
+				new SimpleStringSchema(),
+				KafkaCommon.getConsumerProperties(kafkaInputInfo, kafkaGroupId)
+		);
+		if (readFromStart) {
+			consumer.setStartFromEarliest();
+		}
+
+		StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+		DataStream<String> stream = env.addSource(consumer);
+		if (rebalanceInput) {
+			// TODO: Remove `setParallelism()` once our cluster is actually configured right.
+			DataStreamSource<String> cast = (DataStreamSource<String>)stream;
+			stream = cast.setParallelism(1).rebalance();
```

Comment:
just wanted to double-check we wanted to have this be `1` in version-control

Reply:
My goal is to remove the entire `setParallelism()` call, but then the job will always fail in our cluster while the one K8s node can't communicate with Event Streams.  I plan to wait until ops fixes that before merging down.